modules/snooker/src/main/ts/ephox/snooker/api/TableOperations.ts

```diff
@@ -36,191 +36,230 @@ const outcome = (grid: Structs.RowCells[], cursor: Optional<SugarElement>): Tabl
   cursor
 });
 
-const elementFromGrid = function (grid: Structs.RowCells[], row: number, column: number) {
+const elementFromGrid = (grid: Structs.RowCells[], row: number, column: number) => {
   const rows = GridRow.extractGridDetails(grid).rows;
-  return findIn(rows, row, column).orThunk(function () {
-    return findIn(rows, 0, 0);
-  });
+  return findIn(rows, row, column).orThunk(() =>
+    findIn(rows, 0, 0)
+  );
 };
 
-const findIn = function (grid: Structs.RowCells[], row: number, column: number) {
-  return Optional.from(grid[row]).bind(function (r) {
-    return Optional.from(r.cells[column]).bind(function (c) {
-      return Optional.from(c.element);
-    });
-  });
-};
+const findIn = (grid: Structs.RowCells[], row: number, column: number) =>
+  Optional.from(grid[row]).bind((r) =>
+    Optional.from(r.cells[column]).bind((c) =>
+      Optional.from(c.element)
+    )
+  );
 
-const bundle = function (grid: Structs.RowCells[], row: number, column: number) {
+const bundle = (grid: Structs.RowCells[], row: number, column: number) => {
   const rows = GridRow.extractGridDetails(grid).rows;
   return outcome(grid, findIn(rows, row, column));
 };
 
-const uniqueRows = function (details: Structs.DetailExt[]) {
-  return Arr.foldl(details, function (rest, detail) {
-    return Arr.exists(rest, function (currentDetail) {
-      return currentDetail.row === detail.row;
-    }) ? rest : rest.concat([ detail ]);
-  }, [] as Structs.DetailExt[]).sort(function (detailA, detailB) {
-    return detailA.row - detailB.row;
-  });
-};
-
-const uniqueColumns = function (details: Structs.DetailExt[]) {
-  return Arr.foldl(details, function (rest, detail) {
-    return Arr.exists(rest, function (currentDetail) {
-      return currentDetail.column === detail.column;
-    }) ? rest : rest.concat([ detail ]);
-  }, [] as Structs.DetailExt[]).sort(function (detailA, detailB) {
-    return detailA.column - detailB.column;
-  });
-};
-
-const opInsertRowBefore = function (grid: Structs.RowCells[], detail: Structs.DetailExt, comparator: CompElm, genWrappers: GeneratorsModification) {
+const uniqueRows = (details: Structs.DetailExt[]) =>
```

Comment:
This is very hard to read now (see my previous points about `foldl` readability) and I have to ask why was this changed?

Reply:
Because putting several functions inside of each other on the same line is really hard to read. The newlines are, admittedly, not the easiest, but at least they provide some division between the different functions, as opposed to simply lumping them all onto one big pile. Brackets would make it much easier to read, but we keep arrow-functions naked when possible.